### PR TITLE
Allow to remove 2D editor bounds

### LINKED ISSUE #25616: Remove 2D view limits
**Godot version:**

16d402147b9057c9f7d43ef9b46eb8654e5483cc

**OS/device including version:**

Arch Linux

**Issue description:**

The 2D view will clamp your ability to scroll it depending on where nodes are in it; if you add a node2d, or a control, for example, which extends beyond the view's current limits, the limits will extend such that you can move the view so that you can see these new nodes.

There's a number of issues with this:

1. It doesn't respect the node's size in all cases: if your node just happens to have a really long sprite or collision shape, the view will not compensate for that, only paying attention to the node's position; so, you won't be able to move the view to the point where you can actually see it.
2. It makes it difficult to place elements far away; you can scroll out quite a bit, but for a particularly large level, it becomes cumbersome to deal with the view limits.  If you wanted to place a platform high above the view's limits, you have to work your way up there.
3. It doesn't seem to have a real point; if the idea is that I might get lost somehow, I can just double-click any node in the hierarchy and the view will snap to it.  I don't believe having limits on the view helps me in any way.  Maybe a button which centers the view on the origin would be better.

The current workaround is to throw some dummy control nodes into the scene and set their positions to very high values so the view limits stop being a nuisance, but I think the better solution is to just not have limits to begin with.

### PR DESCRIPTION
**Pull request by KoBeWi**
Resolves #25616

~~EDIT:
Eh, now that I look at it, most of this commit is an indentation change due to additional `if`. Is it acceptable to just use return there?~~

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -3585,81 +3585,86 @@ void CanvasItemEditor::_update_scrollbars() {
...
 
-		h_scroll->show();
-		h_scroll->set_min(MIN(view_offset.x, begin.x));
-		h_scroll->set_max(MAX(view_offset.x, end.x) + screen_rect.x);
-		h_scroll->set_page(screen_rect.x);
+		// Calculate scrollable area
+		v_scroll->set_value(view_offset.y);
+		h_scroll->set_value(view_offset.x);
+	} else {
+		h_scroll->hide();
```
Comment (groud) [inline editor/plugins/canvas_item_editor_plugin.cpp]: I don't think this in good to completely disable the scrollbars. That's the complex part of the problem.
Those bars should still be present if possible, but should adapt depending on where you are.

File: editor/editor_settings.cpp [hunk 2]
```diff
@@ -573,6 +573,7 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	_initial_set("editors/2d/bone_outline_color", Color(0.35, 0.35, 0.35));
 	_initial_set("editors/2d/bone_outline_size", 2);
 	_initial_set("editors/2d/viewport_border_color", Color(0.4, 0.4, 1.0, 0.4));
+	_initial_set("editors/2d/remove_viewport_limits", false);
```
Comment (groud) [inline editor/editor_settings.cpp]: I would go for something opposite. Like "Constraint the editor view to its content." defaulting to `true`.

File: editor/plugins/canvas_item_editor_plugin.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/plugins/canvas_item_editor_plugin.cpp]: Totally changed.

File: editor/editor_settings.cpp (hunk 2 above)
Comment (KoBeWi) [reply editor/editor_settings.cpp]: Is `limit_editor_view` acceptable?

Comment (groud) [reply editor/editor_settings.cpp]: I'd use "constraint", it's more positive. ^^ The thing is that it is ok as an identifier, but the text in the settings isn't really explicit. I would be good to be able to customize it (but that's out of the scope of this PR, so constraint_editor_view should be enough for now).

Comment (akien-mga) [reply editor/editor_settings.cpp]: Note that the verb is "constrain" :)

File: editor/editor_settings.cpp [hunk 3]
```diff
@@ -573,6 +573,7 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	_initial_set("editors/2d/bone_outline_color", Color(0.35, 0.35, 0.35));
 	_initial_set("editors/2d/bone_outline_size", 2);
 	_initial_set("editors/2d/viewport_border_color", Color(0.4, 0.4, 1.0, 0.4));
+	_initial_set("editors/2d/constraint_editor_view", true);
```
Comment (akien-mga) [inline editor/editor_settings.cpp]: Should be "constrain_editor_view", never trust a Frenchman coming up with an English term :P

Comment (groud) [reply editor/editor_settings.cpp]: Damn it, looks like someone is making fun of me...  :p. 
Sorry for the mistake @KoBeWi, we Frenchies tend to add letters where they are not needed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**